jmespath/parser.py

```diff
@@ -371,8 +379,10 @@ def _error_led_token(self, token):
                                     token['type'], 'Invalid token')
 
     def _match(self, token_type=None):
-        if self._current_token() == token_type:
-            self._advance()
+        # inline'd self._current_token()
```

Comment:
It's definitely a micro optimization so it's not a huge improvement:

```
BEFORE INLINE ONLY:

lex_time: 0.40600ms, parse_time: 0.69300ms, search_time: 0.06100ms combined_time: 0.76100ms name: deep_nesting_50

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   382                                               def _match(self, token_type=None):
   383    232000       389235      1.7     51.7          if self._current_token() == token_type:
   384    232000       362920      1.6     48.3              self._advance()
   385                                                   else:


AFTER INLINE ONLY:

lex_time: 0.39600ms, parse_time: 0.68300ms, search_time: 0.05800ms combined_time: 0.74600ms name: deep_nesting_50

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
   382                                               def _match(self, token_type=None):
   383                                                   # inline'd self._current_token()
   384    232000       260609      1.1     50.6          if self._tokens[self._index]['type'] == token_type:
   385                                                       # inline'd self._advance()
   386    232000       254009      1.1     49.4              self._index += 1

```
So for this method, this shows about a 33% improvement in perf.  However, for overall parsing time, the deep_nesting_50 benchmark shows there's about a 2% improvement in total combined parsing time so not huge.

I'm fine with removing this.